panther_lights/src/controller_node.py

```diff
@@ -189,15 +270,76 @@ def _set_image_animation_cb(
         return SetLEDImageAnimationResponse(True, f'Successfully set custom animation')
 
     def _update_animations_cb(self, req: TriggerRequest) -> TriggerResponse:
-        self._update_animations()
+        try:
+            self._update_animations()
+        except (KeyError, FileNotFoundError) as err:
+            return TriggerResponse(False, f'Failed to update animations: {err}')
         return TriggerResponse(True, 'Animations updated successfully')
 
     def _get_animation_by_id(self, animation_id: int) -> PantherAnimation:
+        for animation in self._animations:
+            if animation['id'] == animation_id:
+                return deepcopy(animation['animation'])
```

Comment:
```suggestion
        return deepcopy(self._animations[animation_id])
```
The dict based implementation will simplify this that much